openml/extensions/sklearn/extension.py

```diff
@@ -1608,23 +1610,22 @@ def _prediction_to_probabilities(y: np.ndarray, model_classes: List[Any]) -> pd.
             -------
             pd.DataFrame
             """
-
-            if isinstance(task, (OpenMLClassificationTask, OpenMLLearningCurveTask)):
-                if task.class_labels is not None:
-                    if isinstance(y_train, np.ndarray) and isinstance(task.class_labels[0], str):
-                        # mapping (decoding) the predictions to the categories
-                        # creating a separate copy to not change the expected pred_y type
-                        y = [task.class_labels[pred] for pred in y]
-                else:
-                    raise ValueError("The task has no class labels")
-            else:
+            if not isinstance(task, (OpenMLClassificationTask, OpenMLLearningCurveTask)):
                 return None
 
-            # y: list or numpy array of predictions
+            if task.class_labels is None:
```

Comment:
Still, as far as I can tell this function is only called whenever this condition is already satisfied?

Reply:
I now understand. I think I can do away with the if...else under the exception block and make it cleaner?
And abstract away the checks to the `_prediction_to_probabilities` function.